src/app/file-manager/reducers/downloadables.reducer.ts

```diff
@@ -61,6 +61,21 @@ export function DownloadablesReducer(state = initialState, action: Action): Down
                 })
             };
 
+        case DownloadablesActions.TOGGLE_ITEM_SELECTION:
+
+            const tmp_tog = state.items.filter(item => {
+                if (item.id === action.payload.id) {
+                    item.selected = !item.selected;
+                }
+                return true;
+            });
+
+            return {
+                count: tmp_tog.length,
```

Comment:
These values are computed in the same way for each `return` here. It would be clearer and more concise to use a constructor for `DownloadableState` that accepts `Item[]`, since the other elements `DownloadableState` can be derived from the array argument.

Reply:
DownloadableState is an interface and there is no actual class implementing it. That is why we use it that way in the reducer.  If we really have to use that constructor we'll need to add the class.

Reply:
From my understanding of typescript you can replace the interface with a class of the same name without changing anything else. E.g.
```
export class DownloadableState {
    count: number;
    selectedCount: number;
    items: Item[];
    constructor(items: Item[]) {
      this.items = items;
      this.count = items.length;
      this.selectedCount = items.filter(x => x.selected === true).length;
    }
}
```

and this code becomes
```
return new DownloadableState(tmp_tog);
```

Reply:
yes, we just did it that way. Thanks.